### PR TITLE
[Merged by Bors] - feat: port Data.Rat.Defs and Data.Rat.Basic

### PR DESCRIPTION
**Pull request by kim-em**
fc2ed6f838ce7c9b7c7171e58d78eaf7b438fb0e

- [x] depends on: #996 
- [x] depends on:  #1096

### CODE REVIEW

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#996
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#996~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by Ruben-VandeVelde** [comment]
I think https://github.com/leanprover/std4/commit/560e79852d7640ddf6d884d9cd87ea16ef1d40b7 will help here

**Comment by Ruben-VandeVelde** [comment]
Stopping for today - should be ready for someone else to pick up the errors

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#996~~
* ~~leanprover-community/mathlib4#1096~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

File: Mathlib/Data/Rat/Basic.lean [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+was defined in `Mathlib.Data.Rat.Defs`.
+
+## Main Definitions
+
+- `Rat.field` is the field structure on `ℚ`.
+
+## Implementation notes
+
+We have to define the field structure in a separate file to avoid cyclic imports:
+the `Field` class contains a map from `ℚ` (see `field`'s docstring for the rationale),
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Rat/Basic.lean:29]:
```suggestion
the `Field` class contains a map from `ℚ` (see `Field`'s docstring for the rationale),
```

**Review by jcommelin — APPROVED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**